File: Algorithm/String/Solution/단어 공부 - boj/BOJ1157.c
#include <stdio.h>
#include <ctype.h>
#include <string.h>

int main(void)
{
    const int MAX_SIZE = 1000001;
    const int ALPHABET_COUNT = 26;

    char str[MAX_SIZE];
    fgets(str, MAX_SIZE, stdin);
    int count[ALPHABET_COUNT];
    memset(count, 0, sizeof(count));

    {
        int cnt = 0;
        while (str[cnt] != '\0')
        {
            if (isupper(str[cnt]))
            {
                count[str[cnt] - 'A']++;
            }
            else
            {
                count[str[cnt] - 'a']++;
            }

            cnt++;
        }
    }

    int maxCount = 0;
    char maxChar = '?';

    for (int i = 0; i < ALPHABET_COUNT; ++i)
    {
        if (maxCount < count[i])
        {
            maxCount = count[i];
            maxChar = 'A' + i;
        }
        else if (maxCount == count[i])
        {
            maxChar = '?';
        }
    }

    printf("%c", maxChar);

    return 0;
}

File: Algorithm/String/Solution/단어 공부 - boj/BOJ1157.py
ALPHABET_COUNT = 26
count = [0 for i in range(ALPHABET_COUNT)]
str = input().upper()

for ch in str:
    count[ord(ch) - ord('A')] += 1

maxCount = -1
maxChar = '?'

for i in range(ALPHABET_COUNT):
    if (maxCount < count[i]):
        maxCount = count[i]
        maxChar = chr(ord('A') + i)
    elif (maxCount == count[i]):
        maxChar = '?'

print(maxChar)
